File: src/kamelong/com/JPTI/JPTI/Calendar.java
package kamelong.com.JPTI.JPTI;

import org.json.JSONObject;

/**
 * Created by kame on 2017/06/29.
 */
public class Calendar {
    String name="";

    private static final String NAME="calendar_name";

    public JSONObject makeJSONObject(){
        JSONObject json=new JSONObject();
        try{
            json.put(NAME,name);
        }
        catch(Exception e){
            e.printStackTrace();
        }
        return json;
    }


}

File: src/com/kamelong/JPTI/Calendar.kt
package com.kamelong.JPTI

import java.sql.Connection
import java.sql.ResultSet
import java.util.*
/*
 * Copyright (c) 2019 KameLong
 * contact:kamelong.com
 *
 * This source code is released under GNU GPL ver3.
 */

class Calendar(val id: UUID,val jpti:JPTI){
    constructor(rs: ResultSet, jpti:JPTI):this(
        UUID.fromString(rs.getString("id")),jpti
    ){
        name=rs.getString("calendar_name")
    }
    /**
     * ダイヤ名に相当
     */
    var name:String=""
    fun saveToSQL(conn: Connection){
        val deleteSQL="delete from calendar where id=?"
        val insertSQL="insert into calendar (id,calendar_name) values(?,?)"
        try {
            val ps=conn.prepareStatement(deleteSQL)
            ps.setString(1, id.toString())
            ps.executeUpdate()
        } catch (e: Exception) {
            throw e
        }
        try {
            val ps=conn.prepareStatement(insertSQL)
            ps.setString(1, id.toString())
            ps.setString(2, name)
            ps.executeUpdate()
        } catch (e: Exception) {
            throw e
        }
    }
}